servicetalk-http-api/src/test/java/io/servicetalk/http/api/FormUrlEncodedHttpDeserializerTest.java

```diff
@@ -148,12 +137,29 @@ public boolean hasNext() {
         };
 
         final BlockingIterable<Map<String, List<String>>> deserialized = deserializer
-                .deserialize(headers, formParametersIterable);
+                .deserialize(FE_HEADERS, formParametersIterable);
         deserialized.iterator().close();
 
         assertTrue(isClosed.get(), "BlockingIterable was not closed.");
     }
 
+    @ParameterizedTest(name = "{displayName} [{index}] paramIsNull={0}")
+    @ValueSource(booleans = { true, false })
+    void deserializesNullOrEmptyValues(boolean paramIsNull) {
+        final String separator = paramIsNull ? "" : "=";
+        final String formParameters = String.format("key1%s&key2%s&key2%s", separator, separator, separator);
+
+        final Map<String, List<String>> deserialized = deserializer.deserialize(FE_HEADERS, toBuffer(formParameters));
+        assertThat(deserialized.size(), is(2));
+
+        assertThat(deserialized.get("key1").size(), is(1));
+        assertThat(deserialized.get("key2").size(), is(2));
```

Comment:
Consider:
```
        assertThat(deserialized.get("key1"), hasSize(1));
        assertThat(deserialized.get("key2"), hasSize(2));
```
It generates better exception message in case the assertion fails.